x-pack/plugins/ml/server/routes/job_service.ts

```diff
@@ -27,24 +27,17 @@ import { categorizationExamplesProvider } from '../models/job_service/new_job';
 /**
  * Routes for job service
  */
-export function jobServiceRoutes({ router, mlLicense }: RouteInitialization) {
-  async function hasPermissionToCreateJobs(
-    callAsCurrentUser: IScopedClusterClient['callAsCurrentUser']
-  ) {
-    if (mlLicense.isSecurityEnabled() === false) {
-      return true;
+export function jobServiceRoutes(
+  { router, mlLicense }: RouteInitialization,
+  { resolveMlCapabilities }: JobServiceRouteDeps
+) {
+  async function hasPermissionToCreateJobs(request: KibanaRequest) {
+    const mlCapabilities = await resolveMlCapabilities(request);
+    if (mlCapabilities === null) {
+      throw new Error('resolveMlCapabilities is not defined');
     }
 
-    const resp = await callAsCurrentUser('ml.privilegeCheck', {
-      body: {
-        cluster: [
-          'cluster:admin/xpack/ml/job/put',
-          'cluster:admin/xpack/ml/job/open',
-          'cluster:admin/xpack/ml/datafeeds/put',
-        ],
-      },
-    });
-    return resp.has_all_requested;
+    return mlCapabilities.canCreateJob;
```

Comment:
If I'm logged in as a user without `machine_learning_admin` or `machine_learning_user` roles I still see the ML link in the sample data links on the Kibana Home page. Can this be hidden?

![image](https://user-images.githubusercontent.com/7405507/80007215-decec680-84bd-11ea-8a97-ce552e86362a.png)

Reply:
This should be fixed in a follow PR as it's not a trivial change.
At the point we initialise the same datasets, we do not have the complete capabilities from kibana. 
so the dataset initialisation will need to move, possibly to the plugin start function

Reply:
Thinking about this some more, i'm not sure this is possible. 
This sample dataset links are initialised once per kibana server setup. They are not based on request and so cannot be toggled based on user credentials.

Reply:
Agreed, this doesn't currently seem possible with the way the sample data set links work. I've added it as an extra item to https://github.com/elastic/kibana/issues/64172 so we can follow up the extra changes that are required to the `home` and `ml` plugins.